builtin/logical/ssh/path_config_ca.go

```diff
@@ -112,39 +164,37 @@ func (b *backend) pathConfigCAUpdate(req *logical.Request, data *framework.Field
 		return nil, fmt.Errorf("failed to generate or parse the keys")
 	}
 
-	publicKeyEntry, err := req.Storage.Get("config/ca_public_key")
+	storedPublicKey, err := caKey(req.Storage, "public")
 	if err != nil {
-		return nil, fmt.Errorf("failed while reading ca_public_key: %v", err)
+		return nil, fmt.Errorf("failed to read CA public key: %v", err)
 	}
 
-	privateKeyEntry, err := req.Storage.Get("config/ca_bundle")
+	storedPrivateKey, err := caKey(req.Storage, "private")
 	if err != nil {
-		return nil, fmt.Errorf("failed while reading ca_bundle: %v", err)
+		return nil, fmt.Errorf("failed to read CA private key: %v", err)
 	}
 
-	if publicKeyEntry != nil || privateKeyEntry != nil {
+	if storedPublicKey != "" || storedPrivateKey != "" {
```

Comment:
This is a good point. While storing the keys, if we are able to store the first key and fail to store the next, I'm cleaning up the first one. Thanks for suggesting this.